### ISSUE TITLE
pytraj tries to create files in system directories during install

### ISSUE THREAD

**Issue by drroe** [Amber-MD/pytraj#794]
@hainm tried to compile pytraj today (commit 553ba97d11ec4a5fafdef0af1e93f78102889983) using ` python ./setup.py install`.

Got (showing last few lines only): 
```
g++ -pthread -shared -Wl,-z,relro build/temp.linux-x86_64-2.7/pytraj/Frame.o -L/home/droe/GitHub/cpptraj/lib/ -L/usr/lib64 -lcpptraj -lpython2.7 -o build/lib.linux-x86_64-2.7/pytraj/Frame.so -O0 -ggdb
running install_lib
creating /usr/lib64/python2.7/site-packages/pytraj
error: could not create '/usr/lib64/python2.7/site-packages/pytraj': Permission denied
```
So something is strange with `install_lib` on my system. Any ideas?

**Comment by hainm** [comment]
probably because you're trying to install to ``/usr/lib64/python2.7/site-packages`` (which need ``sudo``?)

A dirty way is try to ``sudo python ./setupy.py install``.
A less dirty way is install python package in your home directory: ``python ./setup.py install --user``

My favorite way is to install Anaconda (http://amber-md.github.io/pytraj/doc/build/html/installation.html#install-anaconda-for-all-python-packages)
Basically, this installation will make a python folder, something like ``/home/drroe/anaconda3/``. Whenever you need to install a new python package (in general), it will be in that folder.

Can you try and let me know if you can do it? thanks

**Comment by drroe** [comment]
I'm just following the README.md file:
```
- easiest and less headache:
    * git clone https://github.com/pytraj/pytraj
    * cd pytraj
    * python ./setup.py install
```
I don't think installs should require sudo by default. What does the `--user` option do? It's not listed by `python ./setup.py --help` or mentioned in the README.

**Comment by hainm** [comment]
> --user

will create .local folder in your home dir
/home/drroe/.local/lib/python3.4/site-packages/pytraj (as example)

Alternatively, you can use ``python ./setup.py install --prefix=your_favorite_dir``. But after doing this, you need to add ``your_favorite_dir``  to ``PYTHONPATH``

```
export PYTHONPATH=your_favorite_dir:${PYTHONPATH}"
```
 
It's python's common practice to install new package, so I am assuming people know. I will update README.md and doc. thanks. :D

**Comment by drroe** [comment]
I see, this is a standard python distutils thing. I'm not so familiar with that. I think updating the README will be helpful for python n00bs like myself. The `--user` option seems to work fine.

**Comment by hainm** [comment]
yeah. I am updating it. will commit in about 15 minutes.

**Comment by swails** [comment]
> What does the --user option do? It's not listed by python ./setup.py --help or mentioned in the README.

It's an option for the `install` subcommand of `setup.py`.  So you are supposed to do `python setup.py install --help`.  This is what happens when you do that for ParmEd:

```
$ python setup.py install --help
Common commands: (see '--help-commands' for more)

  setup.py build      will build the package underneath 'build/'
  setup.py install    will install the package

Global options:
  --verbose (-v)  run verbosely (default)
  --quiet (-q)    run quietly (turns verbosity off)
  --dry-run (-n)  don't actually do anything
  --help (-h)     show detailed help message
  --no-user-cfg   ignore pydistutils.cfg in your home directory

Options for 'install' command:
  --prefix            installation prefix
  --exec-prefix       (Unix only) prefix for platform-specific files
  --home              (Unix only) home directory to install under
  --user              install in user site-package
                      '/Users/swails/.local/lib/python2.7/site-packages'
  --install-base      base installation directory (instead of --prefix or --
                      home)
  --install-platbase  base installation directory for platform-specific files
                      (instead of --exec-prefix or --home)
  --root              install everything relative to this alternate root
                      directory
  --install-purelib   installation directory for pure Python module
                      distributions
  --install-platlib   installation directory for non-pure module distributions
  --install-lib       installation directory for all module distributions
                      (overrides --install-purelib and --install-platlib)
  --install-headers   installation directory for C/C++ headers
  --install-scripts   installation directory for Python scripts
  --install-data      installation directory for data files
  --compile (-c)      compile .py to .pyc [default]
  --no-compile        don't compile .py files
  --optimize (-O)     also compile with optimization: -O1 for "python -O", -O2
                      for "python -OO", and -O0 to disable [default: -O0]
  --force (-f)        force installation (overwrite any existing files)
  --skip-build        skip rebuilding everything (for testing/debugging)
  --record            filename in which to record list of installed files

usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help
```

But I guess you can't really do that for pytraj, because if you do it goes out and tries to do a bunch of stuff before printing the help:

```
[swails@Robin ~/src/pytraj (master) ]$ python setup.py install --help

    Trying to dowload and build libcpptraj for you. (5-10 minutes)
    (check ./cpptraj/ folder after installation)

    To avoid auto-installation
    --------------------------
    Must set CPPTRAJHOME or installing ./cpptraj/ in current folder.

    If you want to manually install `libcpptraj`, you can download cpptraj
    development version from here: https://github.com/Amber-MD/cpptraj

    (    git clone https://github.com/Amber-MD/cpptraj/
         cd cpptraj
         export CPPTRAJHOME=`pwd`
         ./configure -shared gnu
         make libcpptraj    )

    and then go back to pytraj folder:
    python ./setup.py install

    ...
    
Cloning into 'cpptraj'...
remote: Counting objects: 29063, done.
remote: Compressing objects: 100% (81/81), done.
remote: Total 29063 (delta 41), reused 0 (delta 0), pack-reused 28982
Receiving objects: 100% (29063/29063), 73.57 MiB | 3.53 MiB/s, done.
Resolving deltas: 100% (23022/23022), done.
Checking connectivity... done.

Enabling position-independent code for generating shared library.
Using gnu compilers
Installing cpptraj binary to /Users/swails/src/pytraj/cpptraj/bin
Testing C compiler:
  OK
Testing C++ compiler:
  OK
Testing Fortran compiler:
./configure: line 208: ./testp: No such file or directory
  Error: Unable to compile using:
         gfortran -ffree-form  -O3  -DHASBZ2 -DHASGZ -DBINTRAJ -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64  -fPIC
  Please check your compiler settings or configure flags. Error follows:
  --------------------
./configure: line 207: gfortran: command not found
  --------------------


Not using BLAS/LAPACK/ARPACK routines.
Enabling position-independent code for generating shared library.
Using gnu compilers
Installing cpptraj binary to /Users/swails/src/pytraj/cpptraj/bin
Testing C compiler:
  OK
Testing C++ compiler:
  OK
Testing Fortran compiler:
./configure: line 208: ./testp: No such file or directory
  Error: Unable to compile using:
         gfortran -ffree-form  -O3  -DNO_MATHLIB -DHASBZ2 -DHASGZ -DBINTRAJ -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64  -fPIC
  Please check your compiler settings or configure flags. Error follows:
  --------------------
./configure: line 207: gfortran: command not found
  --------------------


Not using BLAS/LAPACK/ARPACK routines.
Enabling position-independent code for generating shared library.
Not using netcdf
Using gnu compilers
Installing cpptraj binary to /Users/swails/src/pytraj/cpptraj/bin
Testing C compiler:
  OK
Testing C++ compiler:
  OK
Testing Fortran compiler:
./configure: line 208: ./testp: No such file or directory
  Error: Unable to compile using:
         gfortran -ffree-form  -O3  -DNO_MATHLIB -DHASBZ2 -DHASGZ -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64  -fPIC
  Please check your compiler settings or configure flags. Error follows:
  --------------------
./configure: line 207: gfortran: command not found
  --------------------

make: *** No rule to make target `config.h', needed by `libcpptraj'.  Stop.


make sure to 'export CPPTRAJHOME=/Users/swails/src/pytraj/cpptraj'
and 'export LD_LIBRARY_PATH=/Users/swails/src/pytraj/cpptraj/lib:$LD_LIBRARY_PATH'
then 'python ./setup.py install'
Common commands: (see '--help-commands' for more)

  setup.py build      will build the package underneath 'build/'
  setup.py install    will install the package

Global options:
  --verbose (-v)  run verbosely (default)
  --quiet (-q)    run quietly (turns verbosity off)
  --dry-run (-n)  don't actually do anything
  --help (-h)     show detailed help message
  --no-user-cfg   ignore pydistutils.cfg in your home directory

Options for 'install' command:
  --prefix            installation prefix
  --exec-prefix       (Unix only) prefix for platform-specific files
  --home              (Unix only) home directory to install under
  --user              install in user site-package
                      '/Users/swails/.local/lib/python2.7/site-packages'
  --install-base      base installation directory (instead of --prefix or --
                      home)
  --install-platbase  base installation directory for platform-specific files
                      (instead of --exec-prefix or --home)
  --root              install everything relative to this alternate root
                      directory
  --install-purelib   installation directory for pure Python module
                      distributions
  --install-platlib   installation directory for non-pure module distributions
  --install-lib       installation directory for all module distributions
                      (overrides --install-purelib and --install-platlib)
  --install-headers   installation directory for C/C++ headers
  --install-scripts   installation directory for Python scripts
  --install-data      installation directory for data files
  --compile (-c)      compile .py to .pyc [default]
  --no-compile        don't compile .py files
  --optimize (-O)     also compile with optimization: -O1 for "python -O", -O2
                      for "python -OO", and -O0 to disable [default: -O0]
  --force (-f)        force installation (overwrite any existing files)
  --skip-build        skip rebuilding everything (for testing/debugging)
  --record            filename in which to record list of installed files

usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help




_____________________                              _____________________
`-._                 \           |\__/|           /                 _.-'
    \                 \          |    |          /                 /  
     \                 `-_______/      \_______-'                 /    
      |                                                          |    
      |                                                          |    
      |                 successfully install pytraj              |
      /                                                          \    
     /_____________                                  ____________\-
                   `----._                    _.----'                
                          `--.            .--'                          
                              `-.      .-'                              
                                 \    /
                                  \  /                                  
                                   \/             

make sure to add /Users/swails/src/pytraj/cpptraj/lib to your LD_LIBRARY_PATH
```

@hainm -- this is *probably* undesirable behavior.

**Comment by swails** [comment]
`python setup.py --help install` works, but it still prints out the deceptive "successfully installed pytraj" banner at the end...

**Comment by hainm** [comment]
oh, dump me. In setup.py, I only check if 'install' in sys.argv and there is not failure. I will fix it.
(similar issue here https://github.com/Amber-MD/pytraj/issues/749)

**Comment by swails** [comment]
The banner is not as bad as trying to clone, configure, and install cpptraj when typing `python setup.py install --help`.  Also, in my particular case, `gfortran` wasn't found so the configure step for cpptraj ended in failure, yet it seems like that failure wasn't caught.

**Comment by hainm** [comment]
I will work on all the things you listed. thanks. will look at amber installation to learn about catching failure.

**Comment by hainm** [comment]
@drroe 

``cpptraj`` need to fix below message too (installing ``libcpptraj`` but cpptraj is saying to throw it into ``bin``)

```
Enabling position-independent code for generating shared library.
Using gnu compilers
Installing cpptraj binary to /Users/swails/src/pytraj/cpptraj/bin
```

**Comment by hainm** [comment]
@drroe

per install issue from your original post, I update pytraj's website here
http://amber-md.github.io/pytraj/doc/build/html/installation.html

(update README.md's link to point to that website).

close this issue and move it to #796
